### ISSUE TITLE
syntax error, unexpected ), expecting (

### ISSUE THREAD

**Issue by brendangregg** [iovisor/bpftrace#527]
This is a minor issue, but should be improved at some point:

```
# bpftrace -e 'BEGIN { printf("%s", foo); }'
1.25: syntax error, unexpected ), expecting (
```

This should error with "foo is unrecognized".

**Comment by mmarchini** [comment]
Should be fixed by #511

**Comment by brendangregg** [comment]
After #511:

```
# bpftrace -e 'BEGIN { printf("%s", foo); }'
Unknown identifier: 'foo'
```

fixed!
